examples/fairmotion_interface.py

```diff
@@ -36,68 +34,77 @@ class FairmotionInterface:
     def __init__(
         self,
         sim,
-        metadata_name=None,
         urdf_path=None,
         amass_path=None,
         bm_path=None,
-        metadata_dir=None,
+        metadata_file=None,
     ) -> None:
         LoggingContext.reinitialize_from_env()
         self.sim = sim
         self.art_obj_mgr = self.sim.get_articulated_object_manager()
-        self.model: habitat_sim.physics.ManagedArticulatedObject = None
-        self.motion: motion.Motion = None
-        self.metadata = {}
-        self.metadata_dir = metadata_dir or METADATA_DIR
-        self.motion_stepper = 0
-        self.setup_default_metadata()
+        self.rgd_obj_mgr = self.sim.get_rigid_object_manager()
+        self.model: Optional[phy.ManagedArticulatedObject] = None
+        self.motion: Optional[motion.Motion] = None
+        self.user_metadata = {}
+        self.last_metadata_file = None
+        self.motion_stepper = 1
+        self.rotation_offset: Optional[mn.Quaternion] = None
+        self.translation_offset: Optional[mn.Vector3] = None
 
         self.key_frames = None
         self.key_frame_models = []
         self.preview_mode = Preview.OFF
         self.traj_ids: List[int] = []
         self.is_reversed = False
+        self.setup_default_metadata()
 
-        # loading from file if given
-        # if file doesn't exist, make a new set of data with
-        # metadata_name as name
-        if metadata_name is not None:
+        if metadata_file:
             try:
-                self.fetch_metadata(metadata_name)
-                self.set_data(name=metadata_name)
-            except Exception:
-                Exception(f"No file with name {metadata_name}, creating new file.")
+                self.fetch_metadata(metadata_file)
+                self.set_data()
+            except FileNotFoundError:
+                FileNotFoundError(
```

Comment:
This does nothing it just constructs the Exception object. If you want to rethrow the exception, `raise` it. If you want to note the error, then use the logger.